### PR TITLE
Expose missing node api

### PR DESCRIPTION
**Pull request by dkozykowski**
### Details:
 - Expose missing APIs from Node Input/Output class

### Tickets:
 - 67433

### CODE REVIEW

File: src/bindings/python/src/pyopenvino/graph/node_input.cpp [hunk 1]
```diff
@@ -91,6 +99,16 @@ void regclass_graph_Input(py::module m) {
...
              )");
+    input.def("replace_source_output",
+              &ov::Input<ov::Node>::replace_source_output,
+              py::arg("new_source_output"),
+              R"(
+                Replaces the source output of this input.
+
+                Returns
+                ----------
+                None
```
Comment (jiwaszki) [inline src/bindings/python/src/pyopenvino/graph/node_input.cpp:110]:
```suggestion
                Replaces the source output of this input.
                
                :param new_source_output: New source output.
                :type new_source_output: openvino.runtime.Output
```
Please use new formatting when adding docs.

File: src/bindings/python/src/pyopenvino/graph/node_output.cpp [hunk 2]
```diff
@@ -13,3 +13,31 @@ namespace py = pybind11;
...
+    output.def("remove_target_input",
+               &ov::Output<ov::Node>::remove_target_input,
+               py::arg("target_input"),
+               R"(
+            Removes a target input from the output referenced by this output handle.
+
+            Returns
+            ----------
+            None
+        )");
```
Comment (jiwaszki) [inline src/bindings/python/src/pyopenvino/graph/node_output.cpp:32]: Please use new formatting when adding docs.

File: src/bindings/python/src/pyopenvino/graph/node_output.cpp [hunk 3]
```diff
@@ -13,3 +13,31 @@ namespace py = pybind11;
...
+    output.def("replace",
+               &ov::Output<ov::Node>::replace,
+               py::arg("replacement"),
+               R"(
+            Replace all users of this value with replacement.
+
+            Returns
+            ----------
+            None
+        )");
```
Comment (jiwaszki) [inline src/bindings/python/src/pyopenvino/graph/node_output.cpp:42]: Update docs to new style. Here `None` means the same as void, there is no need to document default returning value.

**Review by jiwaszki — CHANGES_REQUESTED** [review]

File: src/bindings/python/src/pyopenvino/graph/node_input.cpp (hunk 1 above)
Comment (dkozykowski) [reply src/bindings/python/src/pyopenvino/graph/node_input.cpp]: Updated. Thank you!

File: src/bindings/python/src/pyopenvino/graph/node_input.cpp [hunk 4]
```diff
@@ -91,6 +99,12 @@ void regclass_graph_Input(py::module m) {
                 get_rt_info : RTMap
                     A dictionary of user defined data.
              )");
+    input.def("replace_source_output",
+              &ov::Input<ov::Node>::replace_source_output,
+              py::arg("new_source_output"),
+              R"(
+                Replaces the source output of this input.
+              )");
```
Comment (jiwaszki) [inline src/bindings/python/src/pyopenvino/graph/node_input.cpp:107]: Missing info about parameter!

File: src/bindings/python/src/pyopenvino/graph/node_output.cpp [hunk 5]
```diff
@@ -13,3 +13,23 @@ namespace py = pybind11;
...
+               py::arg("target_input"),
+               R"(
+                Removes a target input from the output referenced by this output handle.
+               )");
+    output.def("replace",
+               &ov::Output<ov::Node>::replace,
+               py::arg("replacement"),
+               R"(
+                Replace all users of this value with replacement.
+               )");
```
Comment (jiwaszki) [inline src/bindings/python/src/pyopenvino/graph/node_output.cpp:34]: Missing info

File: src/bindings/python/src/pyopenvino/graph/node_output.cpp [hunk 6]
```diff
@@ -13,3 +13,23 @@ namespace py = pybind11;
...
+
+template <>
+void def_type_dependent_functions<ov::Node>(
+    py::class_<ov::Output<ov::Node>, std::shared_ptr<ov::Output<ov::Node>>>& output) {
+    output.def("remove_target_input",
+               &ov::Output<ov::Node>::remove_target_input,
+               py::arg("target_input"),
+               R"(
+                Removes a target input from the output referenced by this output handle.
+               )");
```
Comment (jiwaszki) [inline src/bindings/python/src/pyopenvino/graph/node_output.cpp:28]: Missing info

**Review by jiwaszki — CHANGES_REQUESTED** [review]

File: src/bindings/python/src/pyopenvino/graph/node_input.cpp (hunk 4 above)
Comment (dkozykowski) [reply src/bindings/python/src/pyopenvino/graph/node_input.cpp]: Added missing info. Could you take a look if it is fine?

**Review by jiwaszki — APPROVED** [review]

**Pull request merged**